crates/curp/src/client/mod.rs

```diff
@@ -355,6 +389,14 @@ impl ClientBuilder {
         })
     }
 
+    /// Wait for client id
+    async fn wait_for_client_id(&self, state: Arc<state::State>) {
+        while state.client_id() == 0 {
+            tokio::time::sleep(*self.config.propose_timeout()).await;
```

Comment:
waiting for a `propose_timeout()` maybe not a good idea? because `propose_timeout` could be set to a long duration, (that's a user-input configuration value).